### PR TITLE
Pass explicitly version to interactions

### PR DESCRIPTION
**Pull request by ppershing**

### CODE REVIEW

File: src/Ada.ts [hunk 1]
```diff
@@ -313,7 +318,8 @@ export default class Ada {
       stakingBlockchainPointer
     })
 
-    return showAddress(this._send, addressParams);
+    const version = await getVersion(this._send)
```
Comment (refi93) [inline src/Ada.ts:321]: to make it clear what kind of version are we referring to (one may wonder if there can be multiple "versions" of the same call), I'd name the variable appVersion

File: src/interactions/getExtendedPublicKeys.ts [hunk 2]
```diff
@@ -12,11 +12,12 @@ import { ensureLedgerAppVersionAtLeast } from "./getVersion";
 
 export async function getExtendedPublicKeys(
   _send: SendFn,
+  version: Version,
```
Comment (refi93) [inline src/interactions/getExtendedPublicKeys.ts:15]: for the sake of consistency with other interactions, this should be prefixed with an underscore

File: src/interactions/getSerial.ts [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
+import type { Version } from "../types/internal"
 import type { GetSerialResponse } from "../types/public"
 import utils, { assert } from "../utils";
 import { INS } from "./common/ins";
 import { wrapRetryStillInCall } from "./common/retry";
 import { ensureLedgerAppVersionAtLeast } from "./getVersion";
 
-export async function getSerial(_send: SendFn): Promise<GetSerialResponse> {
-  await ensureLedgerAppVersionAtLeast(_send, 1, 2);
+export async function getSerial(_send: SendFn, version: Version): Promise<GetSerialResponse> {
```
Comment (refi93) [inline src/interactions/getSerial.ts:9]: other interactions have the version parameter underscored

Comment (ppershing) [reply src/interactions/getSerial.ts:9]: it is underscored if it is unused

File: src/Ada.ts (hunk 1 above)
Comment (ppershing) [reply src/Ada.ts:321]: noted but for now I will leave it as it is

File: src/interactions/getSerial.ts (hunk 3 above)
Comment (refi93) [reply src/interactions/getSerial.ts:9]: I see

**Pull request merged**